src/__tests__/gameHandlerClass.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { GameHandler } from "../gameHandlerClass";
+import { Player } from "../playerClass";
+import { PokemonFactory } from "../pokemonFactory";
+import { PokemonType } from "../pokemonClass";
+import * as data from "../../assets/poke_data.json";
+
+describe("Test Game Handler class", () => {
+  
+  const pokeData = data.pokemons;
+  const getPlayerOne = (factory: PokemonFactory) => {
+    return new Player("Wojtek", [
+      factory.getPokemonByName("bulbasaur"),
+      factory.getPokemonByName("charmander"),
+      factory.getPokemonByName("pikachu"),
+    ]);
+  } 
+
+  const getPlayerTwo = (factory: PokemonFactory) => {
+    return new Player("Wojtek", [
+      factory.getPokemonByName("bulbasaur"),
+      factory.getPokemonByName("charmander"),
+      factory.getPokemonByName("pikachu"),
+    ]);
+  }
+
+  it("GameHandler.isGameFinished should return false if every player has at least one pokemon", () => {
+    // Given
+    const factory = new PokemonFactory(pokeData);
```

Comment:
I think that factory can be global, like `getPlayerOne` function.
Because using factory should be `pure` - it always returns the same (the same pokemon with same stats) for the same input (pokemon name)